### PR TITLE
Airlock rewrite

### PR DESCRIPTION
**Pull request by Loganbacca**
itshappening.gif

Complete rewrite from the ground up, no caelcode to be found. Details of various fixes in commit logs. All the airlock logic is in airlock_wireless.dm

If you'd like I can make a flow chart showing how it works, also I'll put together a graphic showing how to hook these up in Dream Maker.

In future I'll make them buildable in game, but for now just wanted to get them in.

Are we making changelog entries?

EDIT: also, I'll port this back to the Bay12 repo in the future with the water specific config removed.

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -30,12 +30,14 @@
...
 	var/secured_wires = 0
 	var/datum/wires/airlock/wires = null
+	waterproof = 0
 
 	var/open_sound_powered = 'sound/machines/airlock.ogg'
 	var/open_sound_unpowered = 'sound/machines/airlock_creaking.ogg'
 
 	var/_wifi_id
 	var/datum/wifi/receiver/button/door/wifi_receiver
+	var/_airlock_function
```
Comment (Zuhayr) [inline code/game/machinery/doors/airlock.dm]: Is _var a standard elsewhere in the code?

Comment (Loganbacca) [inline code/game/machinery/doors/airlock.dm]: Not that I'm aware of. I tend to use them as "initial configuration" vars as they are pushed to the top of the variable list in Dream Maker.

**Comment by Zuhayr** [comment]
Honestly I don't know the original code enough to give this a proper review and the parts I do understand are A++, so, let's see how we go.

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (Dennok) [inline code/game/machinery/doors/airlock.dm]: it's brilliant!

File: code/modules/europa/airlock/airlock_pump.dm [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	frequency = null
+
+	var/_airlock_function
+	var/_wifi_id
+	var/datum/wifi/receiver/airlock_pump/wifi_receiver
+
+/obj/machinery/atmospherics/unary/vent_pump/airlock/New()
+	..()
+	if(air_contents)
+		air_contents.volume = ATMOS_DEFAULT_VOLUME_PUMP + 800
```
Comment (Dennok) [inline code/modules/europa/airlock/airlock_pump.dm]: Maybe better use here coefficient like ATMOS_DEFAULT_VOLUME_PUMP * 5 ?

File: code/modules/europa/airlock/airlock_pump.dm [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+//contrary to popular belief this will pump any gas or liquid in the room, not just water
+/obj/machinery/atmospherics/unary/vent_pump/airlock/water_pump
+	name = "airlock water pump"
+	desc = "A gas pump that has been upgraded with a motor powerful enough to pump water in high pressure environments."
+	power_rating = 30000	//30 kW ~ 40 HP - because damnit it's a high pressure underwater environment
+
+/obj/machinery/atmospherics/unary/vent_pump/airlock/water_pump/New()
+	..()
+	if(air_contents)
+		air_contents.volume = ATMOS_DEFAULT_VOLUME_PUMP + 1800
```
Comment (Dennok) [inline code/modules/europa/airlock/airlock_pump.dm]: Maybe better use here coefficient like ATMOS_DEFAULT_VOLUME_PUMP * 10 ?

Comment (Loganbacca) [inline code/modules/europa/airlock/airlock_pump.dm]: Probably, but my next project is to rewrite atmospherics code anyway, so the whole internal volume thing Will probably end up going away.

Comment (Dennok) [inline code/modules/europa/airlock/airlock_pump.dm]: :+1:

**Pull request merged**